From ca524ea2401c3339d45846a99cf8b4dd8ebe4d2e Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 12 Mar 2024 16:37:44 +0100
Subject: [PATCH 0723/1695] drivers: rstctrl: make stm32mp15 controllers
 unpaged

Make reset controller operation handlers unpaged for STM32MP15. These
may be used by the PM suspend sequence.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fd870c0d96899d843f0dfc7fc3334b040b4b8bf3 drivers: rstctrl: make stm32mp15 controllers unpaged]
Change-Id: I6eb015e35a74c01f8249f4bc649ab500b6019bfb
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/366182
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/rstctrl/stm32mp1_rstctrl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/drivers/rstctrl/stm32mp1_rstctrl.c b/core/drivers/rstctrl/stm32mp1_rstctrl.c
index 42841a6ff..d16caef00 100644
--- a/core/drivers/rstctrl/stm32mp1_rstctrl.c
+++ b/core/drivers/rstctrl/stm32mp1_rstctrl.c
@@ -8,6 +8,7 @@
 #include <drivers/stm32mp_dt_bindings.h>
 #include <drivers/stm32mp1_rcc.h>
 #include <io.h>
+#include <keep.h>
 #include <kernel/delay.h>
 #include <kernel/dt.h>
 #include <kernel/dt_driver.h>
@@ -131,6 +132,7 @@ static struct rstctrl_ops stm32_rstctrl_ops = {
 	.assert_level = reset_assert,
 	.deassert_level = reset_deassert,
 };
+DECLARE_KEEP_PAGER(stm32_rstctrl_ops);
 
 static struct rstctrl_ops *stm32_reset_get_ops(unsigned int id __unused)
 {
-- 
2.47.2

